src/main/java/com/puppycrawl/tools/checkstyle/TreeWalker.java

```diff
@@ -150,21 +150,16 @@ protected void processFiltered(File file, FileText fileText) throws CheckstyleEx
         if (!ordinaryChecks.isEmpty() || !commentChecks.isEmpty()) {
             final FileContents contents = getFileContents();
             final DetailAST rootAST = JavaParser.parse(contents);
-            if (!ordinaryChecks.isEmpty()) {
-                walk(rootAST, contents, AstState.ORDINARY);
-            }
-            if (!commentChecks.isEmpty()) {
-                final DetailAST astWithComments = JavaParser.appendHiddenCommentNodes(rootAST);
-                walk(astWithComments, contents, AstState.WITH_COMMENTS);
-            }
-            if (filters.isEmpty()) {
-                addViolations(violations);
-            }
-            else {
-                final SortedSet<Violation> filteredViolations =
-                    getFilteredViolations(file.getAbsolutePath(), contents, rootAST);
-                addViolations(filteredViolations);
-            }
+
+            walk(rootAST, contents, AstState.ORDINARY);
```

Comment:
https://github.com/checkstyle/checkstyle/issues/10053#issuecomment-851670092

Reply:
@shashwatj07 , please reply.

Reply:
I'll try to do what @timurt suggested here https://github.com/checkstyle/checkstyle/issues/10053#issuecomment-852426208

Reply:
it is not what I suggested
I said try to write unit test that will fix pitest without modifying TreeWalker and suggested the way how to do that

Reply:
Yes, I'll follow that. Sorry for being ambiguous.

Reply:
@shashwatj07 , do you have problems to finish this PR ?

Reply:
I have some doubts. I'll discuss them on bdd channel.

Reply:
@shashwatj07 , please create separate sub channel in Discord or lets continue discussion right there.

Whole idea of BDD is reduce of fake/mock/dummy usage as much as possible and use real usecases as much as possible.

> if (filters.isEmpty()) {
removed conditional - replaced equality check with false → SURVIVED 

this one is interesting.
it means that getFilteredViolations should do some work that we can notice by different violations of checkstyle.

Reply:
>  if (!ordinaryChecks.isEmpty()) {
removed conditional - replaced equality check with true → SURVIVED

now I see nuance, if non-comment Checks exists they executes, if not exists, there is no side effect in violations to traverse a tree for nothing, this condition is performance optimization.
But we still can detect this by targeting of checkstyle on one/few big files/folder with config that has no check at all and calculate execution time. If not "instant" - failure. in BDD it would be sounds like "If no Check in config, validation by Checkstyle should be instant on folder".

Reply:
if you succeed kill at least one survival please split it separate PR and let merge it.